### PR TITLE
To be reviewed by Irfan Hassan

### PR DESCRIPTION
**Pull request by Sikandar-Bakht**

### CODE REVIEW

File: Sikandar_Bakht/sprint2/SprintTwoProj/unit_test/test_lambda.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+from aws_cdk import core
+from sprint_two_proj.sprint_two_proj_stack import SprintTwoProjStack
+
+def test_lambda_stack():
+    app = core.App()
+    SprintTwoProjStack(app,"teststack")
+    
+    template=app.synth().get_stack_by_name('teststack').template
+    functions=[resource for resource in template['Resources'].values() if resource['Type']== 'AWS::Lambda::Function']
+    assert len(functions)== 2
```
Comment (muhammadskipq2021) [inline Sikandar_Bakht/sprint2/SprintTwoProj/unit_test/test_lambda.py:11]: Add comments to make it easy for readers to understand the functionality of the code.

File: Sikandar_Bakht/sprint2/SprintTwoProj/sprint_two_proj/sprint_two_proj_stack.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        runtime=lambda_.Runtime.PYTHON_3_6,
+        code=lambda_.Code.from_asset(asset),
+        role=role, timeout=timeout)
+    
+    def create_db_table(self, id, part_key):
+        return db.Table(self, 
+        id, 
+        billing_mode=db.BillingMode.PAY_PER_REQUEST, 
+        partition_key=part_key )
+
```
Comment (muhammadskipq2021) [inline Sikandar_Bakht/sprint2/SprintTwoProj/sprint_two_proj/sprint_two_proj_stack.py:203]: Add for assuming the value for threshold in roll back.

Comment (muhammadskipq2021) [reply Sikandar_Bakht/sprint2/SprintTwoProj/sprint_two_proj/sprint_two_proj_stack.py:203]: As you have added comments after each portion of the code to specify the purpose of each portion. This is very good practice.

File: Sikandar_Bakht/sprint2/SprintTwoProj/sprint_two_proj/sikandar_pipeline_stack.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        unit_test = pipelines.ShellStep("unit_test", commands=["cd ./Sikandar_Bakht/sprint2/SprintTwoProj",  
+                                                               "pip install -r requirements.txt",
+                                                               "pytest unit_test",
+                                                               "pytest integtest"])
+     
+        pipeline.add_stage(Beta, post=[unit_test])
+        
+
+        pipeline.add_stage(Prod, pre=[pipelines.ManualApprovalStep("PromoteToProd")])
+
```
Comment (muhammadskipq2021) [inline Sikandar_Bakht/sprint2/SprintTwoProj/sprint_two_proj/sikandar_pipeline_stack.py:51]: Add comments to make code more effective for readers. Explain part of the pipeline by adding comments as you did in the Main stack file.

**Comment by muhammadskipq2021** [comment]
I review your code. I see most of the things are pretty good like you added comments to separate the subtask in a single code. Here are some points where you can work to make your code more effective and readable for reader.
1-Always add reason while choosing some specific value like the threshold for alarm. 
2-Add comments inline where the reader needs some description. like  Adding roll back you should add comments to describe 3-the purpose of rollback. 
4-Always add comment at the start of file to describe what is done in code. 

All other things are good.

File: Sikandar_Bakht/sprint2/SprintTwoProj/sprint_two_proj/sikandar_pipeline_stack.py (hunk 3 above)
Comment (Sikandar-Bakht) [reply Sikandar_Bakht/sprint2/SprintTwoProj/sprint_two_proj/sikandar_pipeline_stack.py:51]: Noted

File: Sikandar_Bakht/sprint2/SprintTwoProj/unit_test/test_lambda.py (hunk 1 above)
Comment (Sikandar-Bakht) [reply Sikandar_Bakht/sprint2/SprintTwoProj/unit_test/test_lambda.py:11]: Noted

File: Sikandar_Bakht/sprint2/SprintTwoProj/sprint_two_proj/sprint_two_proj_stack.py (hunk 2 above)
Comment (Sikandar-Bakht) [reply Sikandar_Bakht/sprint2/SprintTwoProj/sprint_two_proj/sprint_two_proj_stack.py:203]: Noted. Thanks for input

**Comment by Sikandar-Bakht** [comment]
Thanks for the review, Irfan. I'll add more descriptive comments.

**Pull request merged**